### PR TITLE
add secretmgr read function and example

### PR DESCRIPTION
**Pull request by johnzhanghua**

### CODE REVIEW

**Comment by maxott** [comment]
I'm curious on why you are exposing the SecretService to the user:
```
    secret_service = get_secret_service()
    value = secret_service.read(args.secret_name)
```
Pretty much all the other SDK calls are primarily functional, so I would have expected something like
```
    value = get_secret(args.secret_name)
```
Any particular reason for that, beside future extensions - which then would just be additional methods on the SecretService class?

**Comment by johnzhanghua** [comment]
> I'm curious on why you are exposing the SecretService to the user:
> 
> ```
>     secret_service = get_secret_service()
>     value = secret_service.read(args.secret_name)
> ```
> 
> Pretty much all the other SDK calls are primarily functional, so I would have expected something like
> 
> ```
>     value = get_secret(args.secret_name)
> ```
> 
> Any particular reason for that, beside future extensions - which then would just be additional methods on the SecretService class?

Good point.  It was following the queue service, thought it was the pattern. :)   Happy to simplify it.

**Comment by johnzhanghua** [comment]
Fixed.

File: examples/secret/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	PYTHONPATH=../../src python ${SERVICE_FILE} \
+	--secret-name test-OPENAPI-KEY
+
+DOCKER_TAG=secret-example:latest
+
+docker-build:
+	docker build -t ${DOCKER_TAG} \
+	--build-arg GIT_COMMIT=${GIT_COMMIT} \
+	--build-arg GIT_TAG=${GIT_TAG} \
+	--build-arg TIMEZONE=${BUILD_TZ} \
```
Comment (bclews) [inline examples/secret/Makefile]: `${BUILD_TZ}` is referenced, but it does not seem to be  defined.

File: examples/secret/secret.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ),
+)
+
+def get_secret_service(args: ServiceArgs, logger: logging):
+    """
+    Main function to create tasks and add them to the queue.
+    """
+    value = get_secret(args.secret_name)
+    # ALERT, don't log it in prod, its only for demo purpose
+    logger.info(f"read out value { value }")
```
Comment (bclews) [inline examples/secret/secret.py]: I could see myself accidentally running this in production 🤪
```python
# ALERT, don't log it in prod, its only for demo purpose
logger.info(f"read out value { value }")
```

Can we raise an exception instead? Something like...
```python
try:
    value = get_secret(args.secret_name)
    logger.info("Successfully retrieved secret")  
    .
    .
except (ValueError, SecretRetrievalError) as e:
    logger.error(f"Failed to retrieve secret: {e}")
    raise
```

**Edit:** ...or maybe make the comment bigger and more obnoxious :)

File: examples/secret/secret.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    description="A simple demo service for read secert",
+    parameters = [
+        Parameter(name="secret-name", type=Type.STRING, description="Secret name to read"),
+    ],
+    workflow=PythonWorkflow(
+        min_memory="2Gi", min_cpu="500m", min_ephemeral_storage="4Gi"
+    ),
+)
+
+def get_secret_service(args: ServiceArgs, logger: logging):
```
Comment (bclews) [inline examples/secret/secret.py]: Nit-picky but `get_secret_service` function name is slightly ambiguous. Maybe `retrieve_secret` or `fetch_secret_value` for clarity?

File: src/ivcap_sdk_service/cio/ivcap_io_adapter.py [hunk 4]
```diff
@@ -223,6 +225,36 @@ def get_collection(self, collection_urn: str) -> Collection:
...
+
+            if not response.content:
+                logger.error("Failed to read secret: empty response received.")
+                return ""
+
+            data = response.json()
+            return data["secret-value"]
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to read seceret: {e}")
+            return []
```
Comment (bclews) [inline src/ivcap_sdk_service/cio/ivcap_io_adapter.py]: Should this be `return ""` to match the function definition, which says we are returning a string (ie `-> str`)

File: src/ivcap_sdk_service/cio/ivcap_io_adapter.py [hunk 5]
```diff
@@ -223,6 +225,36 @@ def get_collection(self, collection_urn: str) -> Collection:
...
+            response.raise_for_status()
+
+            if not response.content:
+                logger.error("Failed to read secret: empty response received.")
+                return ""
+
+            data = response.json()
+            return data["secret-value"]
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to read seceret: {e}")
```
Comment (bclews) [inline src/ivcap_sdk_service/cio/ivcap_io_adapter.py]: Small typo `seceret`

**Review by bclews — CHANGES_REQUESTED** [review]
Nice work, I have added a few minor comments.

File: examples/secret/Makefile (hunk 1 above)
Comment (johnzhanghua) [reply examples/secret/Makefile]: 🤣  , some code copied over.

File: examples/secret/secret.py (hunk 3 above)
Comment (johnzhanghua) [reply examples/secret/secret.py]: Is it, its passed in as a 'service' function parameter.  Might just call it 'service'.

File: src/ivcap_sdk_service/cio/ivcap_io_adapter.py (hunk 4 above)
Comment (johnzhanghua) [reply src/ivcap_sdk_service/cio/ivcap_io_adapter.py]: I think it better to raise Exception in that case.

**Review by bclews — APPROVED** [review]

**Pull request merged**